vscode/src/non-stop/FixupController.ts

```diff
@@ -870,15 +870,27 @@ export class FixupController
             return false
         }
 
+        /**
+         * Maximum amount of time to spend formatting.
+         * If the formatter takes longer than this then we will skip formatting completely.
+         */
+        const formattingTimeout = 1000
         const formattingChanges =
-            (await vscode.commands.executeCommand<vscode.TextEdit[]>(
-                'vscode.executeFormatDocumentProvider',
-                document.uri,
-                {
-                    tabSize: document.uri,
-                    insertSpaces: getEditorInsertSpaces(document.uri, vscode.workspace, vscode.window),
-                }
-            )) || []
+            (await Promise.race([
+                vscode.commands.executeCommand<vscode.TextEdit[]>(
+                    'vscode.executeFormatDocumentProvider',
+                    document.uri,
+                    {
+                        tabSize: document.uri,
+                        insertSpaces: getEditorInsertSpaces(
+                            document.uri,
+                            vscode.workspace,
+                            vscode.window
+                        ),
+                    }
+                ),
+                new Promise<void>(resolve => setTimeout(resolve, formattingTimeout)),
```

Comment:
nit: we have a couple of sleep functions in the codebase, should we use one here?

```suggestion
                sleep(formattingTimeout),
```